src/pudl/analysis/plant_parts_eia.py

```diff
@@ -720,6 +771,102 @@ def execute(self, gens_mega):
+        # Filter plant part list to records in 1:m
+        plant_parts_eia_filt = plant_parts_eia.loc[
+            plant_parts_eia.record_id_eia.isin(one_to_many.record_id_eia)
+        ]
+
+        # Get the 'm' generator IDs 1:m
+        one_to_many_single = match_to_single_plant_part(
+            multi_gran_df=plant_parts_eia_filt,
+            ppl=plant_parts_eia,
+            part_name="plant_gen",
+            cols_to_keep=["plant_part"],
+        )[["record_id_eia_og", "record_id_eia", "plant_part_og"]].rename(
+            columns={
+                "record_id_eia": "gen_id",
+                "record_id_eia_og": "record_id_eia",
+                "plant_part_og": "plant_part",
+            }
+        )
+
+        # If any 'duplicate records' actually match to the same generator, these aren't 1:m matches. Drop them and any corresponding non-matches.
+        one_to_many = one_to_many.merge(
+            one_to_many_single, on="record_id_eia", validate="1:m"
+        ).drop_duplicates("gen_id")
+        one_to_many = one_to_many.loc[
+            one_to_many.duplicated(subset="record_id_ferc1", keep=False), :
+        ]
+
+        # For each FERC ID, group and add a unique ferc1_generator_agg_id.
+        # To fix: pandas returns "Cannot interpret 'Int64Dtype()' as a data type"
+        # if we try to make this column and the field an integer
+        one_to_many["ferc1_generator_agg_id"] = (
+            one_to_many.groupby(["record_id_ferc1"]).ngroup().astype("Int64")
+        )
+
+        # Add this column back to the main plant part table to enable later identification of grouped records.
+        plant_parts_eia = plant_parts_eia.merge(
+            one_to_many[["gen_id", "ferc1_generator_agg_id"]],
+            left_on="record_id_eia",
+            right_on="gen_id",
+            how="left",
+            validate="1:1",
+        )
+
+        # Aggregate these records into "plant parts" and concatenate onto plant_part table
+        part_df = PlantPart(part_name).execute(plant_parts_eia)
+        # add in the attributes!
+        for attribute_col in CONSISTENT_ATTRIBUTE_COLS:
+            part_df = AddConsistentAttributes(attribute_col, part_name).execute(
+                part_df, plant_parts_eia
+            )
+        for attribute_col in PRIORITY_ATTRIBUTES_DICT.keys():
+            part_df = AddPriorityAttribute(attribute_col, part_name).execute(
+                part_df, plant_parts_eia
+            )
+        for attribute_col in MAX_MIN_ATTRIBUTES_DICT.keys():
+            part_df = AddMaxMinAttribute(
+                attribute_col,
+                part_name,
+                assign_col_dict=MAX_MIN_ATTRIBUTES_DICT[attribute_col]["assign_col"],
+            ).execute(
+                part_df,
+                plant_parts_eia,
+                att_dtype=MAX_MIN_ATTRIBUTES_DICT[attribute_col]["dtype"],
+                keep=MAX_MIN_ATTRIBUTES_DICT[attribute_col]["keep"],
+            )
```

Comment:
if this is the same as what happens currently in `execute` it might be nice to wrap it in a little `add_attributes` method so you can call it in both places. mostly for future times when we inevitably change the attributes.